src/FSharpVSPowerTools.Core/Utils.fs

```diff
@@ -545,46 +714,52 @@ module Dict =
 [<RequireQualifiedAccess>]
 [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
 module String =
+
+    let inline toCharArray (str:string) = str.ToCharArray()
+
     let lowerCaseFirstChar (str: string) =
-        match str with
-        | null -> null
-        | str ->
-            match str.ToCharArray() |> Array.toList with
-            | [] -> str
-            | h :: t when Char.IsUpper h -> String (Char.ToLower h :: t |> List.toArray)
-            | _ -> str
+        if String.IsNullOrEmpty str 
+         || Char.IsLower(str, 0) then str else 
+        let strArr = toCharArray str
+        match Array.tryHead strArr with
+        | None -> str
+        | Some c  -> 
+            strArr.[0] <- Char.ToLower c
+            String (strArr)
+
 
     let extractTrailingIndex (str: string) =
         match str with
         | null -> null, None
         | _ ->
-            str 
-            |> Seq.toList 
-            |> List.rev 
-            |> Seq.takeWhile Char.IsDigit 
-            |> Seq.toArray 
-            |> Array.rev
-            |> fun chars -> String(chars)
+            let mutable charr = str.ToCharArray() 
+            Array.revInPlace &charr
+            let mutable  digits = Array.takeWhile Char.IsDigit charr
+            Array.revInPlace &digits            
+            String digits
             |> function
                | "" -> str, None
                | index -> str.Substring (0, str.Length - index.Length), Some (int index)
 
-    let trim (value: string) = match value with null -> null | x -> x.Trim()
+    /// Remove all trailing and leading whitespace from the string
+    /// return null if the string is null
+    let trim (value: string) = if isNull value then null else value.Trim()
     
+    /// Splits a string into substrings based on the strings in the array seperators
     let split options (separator: string[]) (value: string) = 
-        match value with null -> null | x -> x.Split(separator, options)
+        if isNull value  then null else value.Split(separator, options)
```

Comment:
you want to call `Null.fill` , `Null.map` instead? otherwise I'm not sure what you mean.
```
    let split2 options (separator: string[]) (value: string) = 
        Null.fill value (value.Split(separator, options))
```